### PR TITLE
Add sample about publish and consume events for event grid

### PR DESCRIPTION
**Pull request by haolingdong-msft**
The sample is migrated from https://github.com/Azure-Samples/event-grid-java-publish-consume-events/tree/master/eventgrid-manage-producer-consumer

### CODE REVIEW

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                        }
+                    }, () -> {
+                        System.out.println("Finished reading events.");
+                    });
+
+            // 9. Publish custom events to the EventGrid.
+            // We create events to send to the service and block until the send has completed.
+            Flux.range(0, NUMBER_OF_EVENTS)
+                .doOnNext(number -> {
+                    String body = String.format("Hello world! Number: %s", number);
```
Comment (weidongxu-microsoft) [inline sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java:208]: Hello world! ???

Comment (haolingdong-msft) [reply sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java:208]: sample in azure-messaging-eventhubs uses 'Hello World' as event body as well, so I keep the name. We can change it to `String body = String.format("Custom event number: %s", number);`

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                .authenticate(credential, profile)
+                .withDefaultSubscription();
+
+            eventHubsManager = EventHubsManager.configure()
+                .withLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                .authenticate(credential, profile);
+
+            eventGridManager = EventGridManager.configure()
+                .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+                .authenticate(credential, profile);
```
Comment (weidongxu-microsoft) [inline sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java:84]: `EventHubsManager` seems not needed? `AzureResourceManager` should contains it.

Also you can see if they can share a same `HttpClient` instance.

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+            e.printStackTrace();
+        }
+    }
+
+    private static void runSample() {
+        try {
+
+            // 1. Create a resource group.
+            ResourceGroup resourceGroup =
+                resourceManager.resourceGroups().define(resourceGroupName).withRegion(Region.US_EAST).create();
```
Comment (weidongxu-microsoft) [inline sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java:99]: Should use the `region` here?

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+                .create();
+
+            System.out.println("EventHub namespace created with name " + namespace.name());
+
+            // 2.2 Create event hub.
+            EventHub eventHub = resourceManager.eventHubs()
+                .define(eventHubName)
+                .withExistingNamespace(resourceGroupName, eventHubNamespace)
+                .withNewManageRule(eventHubRuleName)
+                .withPartitionCount(1)
```
Comment (weidongxu-microsoft) [inline sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java:120]: Maybe you can add some comment about partition size.

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+            // This shouldn't happen, but if we are unable to get the partitions within the timeout period.
+            if (firstPartition == null) {
+                firstPartition = "0";
+            }
+
+            // 8.2 Subscribe to the coming events from event grid
+            // We use CountDownLatch to wait for the coming events
+            CountDownLatch countDownLatch = new CountDownLatch(NUMBER_OF_EVENTS);
+            consumer.receiveFromPartition(firstPartition, EventPosition.latest())
+                .subscribe(partitionEvent -> {
```
Comment (weidongxu-microsoft) [inline sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java:185]: Ok, `.subscribe` is fine. This part actually happen after the `sendEvent`. You cannot block here.

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java (hunk 5 above)
Comment (haolingdong-msft) [reply sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java:185]: This consuming part is referred to this sample: https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/eventhubs/azure-messaging-eventhubs/src/samples/java/com/azure/messaging/eventhubs/ConsumeEvents.java

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java (hunk 2 above)
Comment (haolingdong-msft) [reply sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java:84]: updated

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java (hunk 3 above)
Comment (haolingdong-msft) [reply sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java:99]: updated

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java (hunk 4 above)
Comment (haolingdong-msft) [reply sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java:120]: updated

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+# Azure Event Grid Publish/Consume Samples for Java
+
+This contains Java samples for publishing events to Azure Event Grid and consuming events from Azure Event Grid. 
+It also contains a set of management samples that demonstrates how to manage topics and event subscriptions using Java code.
+
+## Features
+
+These samples demonstrate the following features:
+
+- How to create an event grid topic using Java
```
Comment (qwordy) [inline sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md:20]:
```suggestion
- How to create an event grid topic using Java.
```

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+- How to create an event subscription to a topic using Java.
+- How to create an event hub using Java.
+- How to publish events to Azure Event Grid using Java.
+- How to consume events delivered by Azure Event Grid through an Azure Event Hub consumer using Java.
+
+## Getting started
```
Comment (qwordy) [inline sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md:26]:
```suggestion
## Getting Started
```

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+Please refer it to add dependency and configure authentication environment variables.
+
+For details on including this dependency in other build tools (Gradle, SBT, etc), refer [here](https://search.maven.org/artifact/com.azure/azure-core).
+
+## Sample Details
+
+Azure event grid publish and consume sample is in 'sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java',
+it has the `main()` function and can be run directly.
+
+The sample demonstrate following operations:
```
Comment (qwordy) [inline sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md:38]:
```suggestion
The sample demonstrates following operations:
```

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+5. Retrieve the event grid client connection key.
+6. Create an event grid publisher client: `EventGridPublisherClient`.
+7. Create an event hub consumer client: `EventHubConsumerAsyncClient`. 
+8. Subscribe to coming events from event grid using `EventHubConsumerAsyncClient`.
+9. Publish custom events to the event grid using `EventGridPublisherClient`.
+10. Clean up the resources created above.
+
+## Key concepts
+
+Key concepts are explained in detail [here][EVENTGRID_README_KEY_CONCEPTS].
```
Comment (qwordy) [inline sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md:53]: You can add link to `detail` and remove `here`.

Comment (haolingdong-msft) [reply sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md:53]: This is for consistency purpose, readme for other samples keep the format like: Key concepts are explained in detail [here][EVENTGRID_README_KEY_CONCEPTS].

File: sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md (hunk 8 above)
Comment (haolingdong-msft) [reply sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/README.md]: updated

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/eventgrid/azure-resourcemanager-eventgrid/pom.xml [hunk 10]
```diff
@@ -51,6 +51,38 @@
...
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-test</artifactId>
+      <version>1.7.0</version> <!-- {x-version-update;com.azure:azure-core-test;dependency} -->
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-identity</artifactId>
+      <version>1.3.5</version> <!-- {x-version-update;com.azure:azure-identity;dependency} -->
```
Comment (haolingdong-msft) [inline sdk/eventgrid/azure-resourcemanager-eventgrid/pom.xml:65]: update version when core is released

**Pull request merged**